dev guide release post (#1204)
ropensci/roweb3: content/blog/2026-01-09-devguide-release-1.0.0/index.md

---
title: "rOpenSci Dev Guide 1.0.0: Trilingual and Improved"
author:
  - Maëlle Salmon
  - Mark Padgham
  - Noam Ross
date: '2026-01-09'
slug: devguide-1.0.0
tags:
  - Software Peer Review
  - dev guide
  - multilingual
description: "Updates in version 1.0.0 of the online book 'rOpenSci Packages: Development, Maintenance, and Peer Review'"
twitterImg: img/blog-images/2019-05-16-dev-guide-update/cover.png
output: 
twitterAlt: cover of rOpenSci dev guide, showing a package production line with small humans discussing, examining and promoting packages
params:
  doi: "10.59350/pthhe-b9v52"
---

rOpenSci Software Peer Review's guidance is gathered in an [online book](https://devguide.ropensci.org/booknews) that keeps improving!
This blog post summarises what's new in our Dev Guide 1.0.0, with all changes listed in the [changelog](https://devguide.ropensci.org/booknews.html).

## Now available in Portuguese!

Our guide is now trilingual ([English](https://devguide.ropensci.org/index.html), [Spanish](https://devguide.ropensci.org/es/index.es.html), [Portuguese](https://devdevguide.netlify.app/pt/index.pt))!

> Find more about the awesome translation project to Portuguese in our [blog post](/blog/2025/11/25/translation-devguide-pt/).

Reviewer: ```suggestion
Find more about the awesome Portuguese translation project, initiated and powered by our Lusophone members in our [blog post](/blog/2025/11/25/translation-devguide-pt/).
```

OK, this is part that I really like the word "Lusophone", but I wanted a little extra highlight on the fact that the community started this.

- New requirement to not call the [default branch "master"](https://devdevguide.netlify.app/pkg_building#version-control#:~:text=not%20be%20master).

- Removed requirement to use a `codemeta.json` file, as it is no longer relevant.

## Editor Guide, Author Guide & Reviewer Guide

The guides that live in our guide. :smile_cat:

The whole [editor guide](https://devdevguide.netlify.app/softwarereview_editor) was has been restructured to follow the typical flow of submissions,
and to better explain how to use the [software-review dashboard](https://dashboard.ropensci.org/).
We added a section on challenges, and documented how to put the system on vacation (which we generally do over the new year period).

Likewise, we improved the organization and content of the [author guide](https://devdevguide.netlify.app/softwarereview_author) (thanks to [Alec Robitaille](https://github.com/robitalec) and [Joan Maspons](https://github.com/jmaspons)).

In the [reviewer guide](https://devdevguide.netlify.app/softwarereview_reviewer), we removed the link to the defunct Mozilla Review guide in favor of explicit items.

## Packaging Best Practices

> In the packaging guide (another guide within the guide!), we added a minimal mention of criteria for choosing [example datasets](https://devdevguide.netlify.app/pkg_building#example-datasets). 

Reviewer: ```suggestion
In the packaging guide (another guide within the guide!), we added guidance for choosing [example datasets](https://devdevguide.netlify.app/pkg_building#example-datasets). 
```

Furthermore, we created a section for [Packages wrapping external software](https://devdevguide.netlify.app/pkg_building#external-software).
The [licencing section](https://devdevguide.netlify.app/pkg_building#licence) now explicitly requires acknowledging authors of bundled code.
Last but not least, the [section about dependencies](https://devdevguide.netlify.app/pkg_building#pkgdependencies) recommends checking the development status of dependencies.

The whole book now mentions the [Air CLI](https://posit-dev.github.io/air/) every time it mentions the styler package, as Air can be viewed as styler's successor.

> In the chapter about Package evolution, we added how to [deprecate *data*](https://devdevguide.netlify.app/maintenance_evolution#data-deprecate), and explained the [drawbacks of renaming a widely-used package](https://devdevguide.netlify.app/maintenance_evolution#renaming-packages).

Reviewer: ```suggestion
In the chapter about Package evolution, we added guidance on [deprecating *data*](https://devdevguide.netlify.app/maintenance_evolution#data-deprecate), and explained the [drawbacks of renaming a widely-used package](https://devdevguide.netlify.app/maintenance_evolution#renaming-packages).
```

We're actively working on our [babeldown package](https://docs.ropensci.org/babeldown/) to create and update translations using the DeepL API.

In the [dev guide](https://github.com/ropensci/dev_guide/pull/940/files) itself, tools useful for internationalizing _packages_ are mentioned: potools, the experimental rhelpi18n package, selecting a language for a pkgdown website.

## Policy Updates

We made some changes to rOpenSci policies and scope:

- New category for [rOpenSci internal and peer-review tools](https://devdevguide.netlify.app/softwarereview_policies#package-categories).

- New explicit rule to only [submit one package at a time](https://devdevguide.netlify.app/softwarereview_author#lifecycle).

- New requirement to not call the [default branch "master"](https://devdevguide.netlify.app/pkg_building#version-control).

## Editor Guide, Author Guide & Reviewer Guide

The guides that live in our guide. :smile_cat:

The whole [editor guide](https://devdevguide.netlify.app/softwarereview_editor) was revamped so that its structure follows the flow of submissions,
so that it better explains how to use the [software-review dashboard](https://dashboard.ropensci.org/).

> We added a section on challenges, and documented how to put the system on vacation.

Reviewer 2: ```suggestion
We added a section on challenges, and documented how to put the system on vacation (which we generally do over the new year period).
```

Likewise, we improved the organization and content of the [author guide](https://devdevguide.netlify.app/softwarereview_author) (thanks to [Alec Robitaille](https://github.com/robitalec) and [Joan Maspons](https://github.com/jmaspons)).

In the [reviewer guide](https://devdevguide.netlify.app/softwarereview_reviewer), we removed the link to the defunct Mozilla Review guide in favor of explicit items.

## Packaging Best Practices

In the packaging guide (another guide within the guide!), we added a minimal mention of criteria for choosing [example datasets](https://devdevguide.netlify.app/pkg_building#example-datasets). 
Furthermore, we created a section for [Packages wrapping external software](https://devdevguide.netlify.app/pkg_building#external-software).

> Besides, the [licencing section](https://devdevguide.netlify.app/pkg_building#licence) now explictly state one should acknowledge authors of bundled code.

Reviewer 2: ```suggestion
The [licencing section](https://devdevguide.netlify.app/pkg_building#licence) now explicitly requires acknowledging authors of bundled code.
```

## Testing guidance

We updated our [testing guidance](https://devdevguide.netlify.app/pkg_building.html#testing) with

- a mention of tinytest as an alternative to testthat;
- a note on top-level code in tests written with testthat.

## Package Documentation

We documented :wink: our documentation building system a bit better, thanks in particular to [Alasdair Warwick ](https://github.com/rmgpanw):

- More details on how to [safeguard](https://devdevguide.netlify.app/pkg_ci#rodocsci) docs building for rOpenSci packages;
- Updated [math guidance](https://devdevguide.netlify.app/pkg_building.html#mathjax) for pkgdown based on pkgdown's update.

## Metadata & Package Information

We documented more ways to acknowledge contributors:

-  in the [authorship](https://devdevguide.netlify.app/pkg_building.html#authorship) section of the packaging guide, [ROR IDs](/blog/2025/05/09/ror/);

- in the chapter about [collaboration](https://devdevguide.netlify.app/maintenance_collaboration.html#attributions), the [allcontributors R package](https://docs.ropensci.org/allcontributors/).

## Conclusion

> In this post we summarized the changes incorporated into our book [“rOpenSci Packages: Development, Maintenance, and Peer Review”](https://devguide.ropensci.org) since the last release.

Reviewer 2: ```suggestion
In this post we summarized the changes in the latest version of our book [“rOpenSci Packages: Development, Maintenance, and Peer Review”](https://devguide.ropensci.org).
```
